docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+be dispatched to the customers choice of notifier(Slack, email, hipchat).
+
+   .. code-block:: yaml
+
+       apiVersion: v1
+       kind: ConfigMap
+       metadata:
+         namespace: metalk8s-monitoring
+         name: <example-name>
+       data:
+
+        channel.smtp.smtp_smarthost: 'smtp.gmail.com:587'
+        channel.smtp.smtp_from: 'foo@bar.com'
+        channel.smtp.smtp_auth_username: 'foo@bar.com'
+        channel.smtp.smtp_auth_password: 'barfoo'
+
+        channel.slack.slack_api_url: 'https://hooks.slack.com/services/abc123'
+
+        channel.hipchat.hipchat_auth_token: '1234556789'
+        channel.hipchat.hipchat_url: 'https://hipchat.foobar.org/'
+
+        # route.group_wait: '30s'
+        # route.repeat_interval: '15m'
+        # route.group_interval: '5m'
+        route.receiver: 'default'
+
+        routes.match.severity: 'email'
+        routes.receiver: 'default'
+        routes.match.severity: 'slack'
+        routes.match.receiver: 'slack_alert'
+
+        receivers.name: 'default'
+        receivers.email_configs.to: 'foo@bar.com'
+
+        receivers.name: 'slack_alert'
+        receivers.slack_configs.channel: '#devops'
+        receivers.slack_configs.send_resolved: 'true'
+
+Iteration 1
+~~~~~~~~~~~
+
+- Define and deploy new namespaced bounded ConfigMap KV stores that will hold
+  user configurations as listed in the requirements
+- Template and render Deployment and Pod manifests that will make use of
+  this user customizable configurations using default pillar values
+- Provide a CLI tool(Bash script) for changing any of the user configurations:
+
+    - Count of replicas for chosen Deployments(Prometheus)
+    - Adding/modifying/removing a Dex authentication connector(OpenLDAP, AD and
+      staticUser store)
+    - Adding/modifying/removing Alertmanager notification configuration
+
+Iteration 2
+~~~~~~~~~~~
+
+- Provide a UI interface that allows CRUD operations on all user customizable
+  settings based on the above requirements
+- Provide a UI interface for adding, updating and deleting service specific
+  configurations for example Dex-LDAP  connector integration.
+- Provide a UI interface for listing MetalK8s available/supported Dex
+  authentication Connectors
+- Provide a UI interface for enabling or disabling Dex authentication
+  connectors(LDAP, Active Directory, StaticUser store)
+- Provide a UI interface for changing the number of replicas for a chosen set
+  of MetalK8s deployments(Prometheus, ....)
+- Add a UI interface for listing Alertmanager notification systems MetalK8s
+  will support(Slack, email, hipchat)
+- Provide a UI interface for adding, modifying and deleting Alertmanager
+  configurations from the listing above
+
+Documentation
+-------------
+
+In the Operational Guide:
+
+* Document how to customize or change any given setting using the CLI tool
+* Document how to customize or change any given setting using the UI interface
```

Comment:
OK, maybe a "common doc" for CLI / UI. But we'll also need this info next to the services we want our users to configure (e.g. show how to configure Dex in the section/page dedicated to Dex)